src/v/wasm/BUILD

```diff
@@ -47,6 +46,9 @@ redpanda_cc_library(
     hdrs = [
         "ffi.h",
     ],
+    implementation_deps = [
+        "//src/v/utils:vint",
```

Comment:
since this is public anyway, the goal here is something other than visibility, like to avoid recompilation in some cases? i guess in this case the idea is that if vint changes then downstream dependencies of ffi don't recompile?

i guess for something like that layering check wasn't sufficient because you could opt-out of it if you wanted. in this case downstream deps need a stronger guarnatee that the ffi.h cannot be seen at all.

Reply:
> i guess in this case the idea is that if vint changes then downstream dependencies of ffi (or is it vint?) don't recompile?

Yes exactly. Anything only referenced in an implementation file should be in `implementation_deps`, it allows for Bazel to provide stronger caching